auth_config.py

```diff
@@ -0,0 +1,11 @@
+from pydantic import Field, BaseModel
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+class AuthProfile(BaseModel):
+    username: str
+    password: str
```

Comment:
Hmm, to me it seems like Quetz uses a `X-API-Key` header instead of the URL scheme you described:

https://github.com/mamba-org/quetz?tab=readme-ov-file#create-a-proxy-channel

Reply:
It seems like the `/t/<api-key>` URL is also supported though: https://github.com/mamba-org/quetz/issues/450#issuecomment-914169520

If both works, I will use `X-API-Key` because this is easier to implement

Reply:
Other authentication backends implemented in dfdddca5ae2576d779d6b38696487d6d2118791d

I tested manually that each of them work (of course, a true testing setup would be better...)

Feel free to resolve this if you think that using `X-API-Key` header is okay (which probably excludes Anaconda private channel support?)